src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -131,8 +163,13 @@ public int IndexOf(T item)
         /// <param name="item">The object to insert into the <see cref="ReadOnlyCollectionBuilder{T}"/>.</param>
         public void Insert(int index, T item)
         {
-            if (index > _size)
+            // Note that insertions at the end are legal.
+            // Following trick can reduce the range check by one
+            if ((uint)index > (uint)_size)
+            {
                 throw new ArgumentOutOfRangeException(nameof(index));
+            }
+            Contract.EndContractBlock();
```

Comment:
Nit: `Contract.EndContractBlock` calls are not needed in corefx-- you can delete these.